flytekit/core/type_engine.py

```diff
@@ -838,6 +919,7 @@ def get_transformer(cls, python_type: Type) -> TypeTransformer[T]:
 
             python_type = args[0]
 
+        # Step 2
```

Comment:
Scope creep: Referring to the algorithm description in the doc string, the numbering was off here.